vim-patch:9.1.{1214,1217,1219}: matchfuzzy() "camelcase"

glepnir opened the pull request:
Problem:  When searching for "Cur", CamelCase matches like "lCursor" score
          higher than exact prefix matches like Cursor, which is
          counter-intuitive (Maxim Kim).
Solution: Add a 'camelcase' option to matchfuzzy() that lets users disable
          CamelCase bonuses when needed, making prefix matches rank higher.
          (glepnir)

closes: vim/vim#16797

https://github.com/vim/vim/commit/28e40a7b55ce471656cccc2260c11a29d5da447e

glepnir committed:
vim-patch:9.1.1214: matchfuzzy() can be improved for camel case matches

Problem:  When searching for "Cur", CamelCase matches like "lCursor" score
          higher than exact prefix matches like Cursor, which is
          counter-intuitive (Maxim Kim).
Solution: Add a 'camelcase' option to matchfuzzy() that lets users disable
          CamelCase bonuses when needed, making prefix matches rank higher.
          (glepnir)

fixes: vim/vim#16504
closes: vim/vim#16797

https://github.com/vim/vim/commit/28e40a7b55ce471656cccc2260c11a29d5da447e

Co-authored-by: glepnir <glephunter@gmail.com>

zeertzjq commented on src/nvim/eval.lua:
```diff
@@ -7148,6 +7148,9 @@ M.funcs = {
           limit	Maximum number of matches in {list} to be
       		returned.  Zero means no limit.
+          camelcase	Use enhanced camel case scoring making results
+                better suited for completion related to
+                programming languages. Default is v:true
```
Wrong indent
```suggestion
      		better suited for completion related to
      		programming languages. Default is v:true
```

zeertzjq commented on src/nvim/eval.lua:
```diff
@@ -7148,6 +7148,9 @@ M.funcs = {
           limit	Maximum number of matches in {list} to be
       		returned.  Zero means no limit.
+          camelcase	Use enhanced camel case scoring making results
+		better suited for completion related to
+		programming languages. Default is v:true
```
There should be 6 spaces before the first TAB

glepnir replied:
thanks

przepompownia commented:
The typehin from https://github.com/neovim/neovim/pull/32985#issuecomment-2736435969 should be placed here rather than directly in #32985.

zeertzjq committed:
vim-patch:9.1.1217: tests: typos in test_matchfuzzy.vim

Problem:  tests: typos in test_matchfuzzy.vim (after 9.1.1214).
Solution: Fix the typos.  Consistently put the function call on the
          second line in assertions for camelcase (zeertzjq).

closes: vim/vim#16907

https://github.com/vim/vim/commit/85627732e0f5f4e31137fc12bcb809418a4f683f

zeertzjq committed:
vim-patch:9.1.1219: Strange error with wrong type for matchfuzzy() "camelcase"

Problem:  Strange error with type for matchfuzzy() "camelcase".
Solution: Show the error "Invalid value for argument camelcase" instead
          of "Invalid argument: camelcase" (zeertzjq).

Note that using tv_get_string() will lead to confusion, as when the
value cannot be converted to a string tv_get_string() will also give an
error about that, but "camelcase" takes a boolean, not a string.  Also
don't use tv_get_string() for the "limit" argument above.

closes: vim/vim#16926

https://github.com/vim/vim/commit/c4815c157b27923001e44bfd241fb540bf1fb518
